From 5a57a8b4425ecfcfaf01bc280f58a05b03cd0c72 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Sun, 20 Oct 2024 23:26:19 +0800
Subject: [PATCH 08/12] Revert "mm: align larger anonymous mappings on THP
 boundaries"

This reverts commit efa7df3e3bb5da8e6abbe37727417f32a37fba47.

This mitigates https://bugzilla.kernel.org/show_bug.cgi?id=219366 until a proper fix is found.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 mm/mmap.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/mm/mmap.c b/mm/mmap.c
index 18fddcce0..d84d6dd87 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -1952,10 +1952,6 @@ __get_unmapped_area(struct file *file, unsigned long addr, unsigned long len,
 
 	if (get_area) {
 		addr = get_area(file, addr, len, pgoff, flags);
-	} else if (IS_ENABLED(CONFIG_TRANSPARENT_HUGEPAGE)) {
-		/* Ensures that larger anonymous mappings are THP aligned. */
-		addr = thp_get_unmapped_area_vmflags(file, addr, len,
-						     pgoff, flags, vm_flags);
 	} else {
 		addr = mm_get_unmapped_area_vmflags(current->mm, file, addr, len,
 						    pgoff, flags, vm_flags);
-- 
2.47.0

